File: software/examples/java/ExampleSimple.java
import com.tinkerforge.BrickletHumidity;
import com.tinkerforge.IPConnection;

public class ExampleSimple {
	private static final String host = "localhost";
	private static final int port = 4223;
	private static final String UID = "ABC"; // Change to your UID
	
	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the commnents below
	public static void main(String args[]) throws Exception {
		// Create connection to brickd
		IPConnection ipcon = new IPConnection(host, port); // Can throw IOException
		BrickletHumidity hum = new BrickletHumidity(UID); // Create device object

		// Add device to IP connection
		ipcon.addDevice(hum); // Can throw IPConnection.TimeoutException
		// Don't use device before it is added to a connection

		// Get current humidity (unit is %RH/10)
		int humidity = hum.getHumidity(); // Can throw IPConnection.TimeoutException

		System.out.println("Relative Humidity: " + humidity/10.0 + " %RH");

		System.console().readLine("Press key to exit\n");
		ipcon.destroy();
	}
}

File: software/examples/c/example_simple.c

#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_humidity.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

int main() {
	// Create IP connection to brickd
	IPConnection ipcon;
	if(ipcon_create(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not create connection\n");
		exit(1);
	}

	// Create device object
	Humidity h;
	humidity_create(&h, UID); 

	// Add device to IP connection
	if(ipcon_add_device(&ipcon, &h) < 0) {
		fprintf(stderr, "Could not connect to Bricklet\n");
		exit(1);
	}
	// Don't use device before it is added to a connection

	// Get current humidity (unit is %RH/10)
	uint16_t humidity;
	if(humidity_get_humidity(&h, &humidity) < 0) {
		fprintf(stderr, "Could not get value, probably timeout\n");
		exit(1);
	}

	printf("Relative Humidity: %f %%RH\n", humidity/10.0);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon);
}
